experimental/dds/tree/src/EditLog.ts

```diff
@@ -425,27 +469,24 @@ export class EditLog<TChange = unknown> extends TypedEventEmitter<IEditLogEvents
 		info?: EditSequencingInfo,
 		minSequenceNumber: number = 0
 	): void {
-		assert(
-			minSequenceNumber >= this.minSequenceNumber,
-			'Sequenced edits should carry a monotonically increasing min number'
-		);
-		// The new minSequenceNumber indicates that no future edit will require information from edits with a smaller or equal seq number
-		// for its resolution.
-		this._minSequenceNumber = minSequenceNumber;
-		// TODO:#57176: Increment maximumEvictableIndex to reflect the fact we can now evict edits with a sequenceNumber lower or equal to
-		// it. Note that this will change the meaning of our 'InSession' APIs so we should make sure to rename them at the same time.
-		// The code might look like this:
-		// while (this.maximumEvictableIndex + 1 < this.indexOfFirstEditInSession) {
-		// 	const nextEdit = this.getEditInSessionAtIndex(this.maximumEvictableIndex + 1);
-		// 	const nextEditInfo = this.getOrderedEditId(nextEdit.id) as SequencedOrderedEditId;
-		// 	if (
-		// 		nextEditInfo.sequenceInfo !== undefined &&
-		// 		nextEditInfo.sequenceInfo.sequenceNumber > minSequenceNumber
-		// 	) {
-		// 		break;
-		// 	}
-		// 	++this.maximumEvictableIndex;
-		// }
+		// The index of the sequenced edit to add
+		const index = this._earliestAvailableEditIndex + this.numberOfSequencedEdits;
+
+		if (info !== undefined) {
+			assert(
+				minSequenceNumber >= this._minSequenceNumber,
+				'Sequenced edits should carry a monotonically increasing min number'
+			);
+			this._minSequenceNumber = minSequenceNumber;
+
+			// Calculate the index of the minSequenceNumber by comparing it to the index of this sequenced edit
+			const sequenceNumberDifference = info.sequenceNumber - minSequenceNumber;
+			const minSequenceIndex = sequenceNumberDifference > 0 ? index - sequenceNumberDifference : index;
```

Comment:
This is calculating the size of the current collab window, and then using that as an offset from the new index to find the oldest index. But, similar to that earlier change below, aren't those different units? Sequence numbers are incremented for every op, but indexes are only incremented for _shared tree edit_ ops. What happens if somebody has a SharedMap and a SharedTree in the same container that are interleaving ops? Won't the size of the collab window here include some of the ops made by the SharedMap, and thus will be too large? I might be wrong about this but we should check that.

I think we need to remember the index that is associated with each sequence number, i.e. have a map `sequence number -> index` that is appended to every time this method is called. Then, to get the minIndex, you query the map with the current minSequenceNumber. I think the map begins with a single entry [0, 0] to boostrap it. Except that, it should be a sparse map, because we might not have an entry for every sequence number. So your map might look like:

[0, 0]
[4, 1]
[5, 2]
[7, 3]

and if you query for "6" you get "2". The values are always increasing by exactly one, and the keys are also increasing, but they might have gaps. And then we can remove old entries from the map so it doesn't grow unbounded; any keys less than the current min sequence number can be deleted. Taylor's `AppendOnlySortedMap` is almost perfect for this, it just doesn't have a way to delete old things. You could probably just use an array and an index offset. 

Am I overcomplicating things here? This feels like too much but I can't think of how it could be simpler given that sequence numbers are "sparse". I'm pretty sure they are; if I run a random tree unit test the first sequence number I get is "5", for example. Maybe you can think of a trick to reduce the complexity though.

Reply:
Done!